src/main/java/string/Exercise01.java

```diff
@@ -0,0 +1,39 @@
+package string;
+
+public class Exercise01 {
+    public boolean isPalindrome(String str) {
+        int left = 0;
+        int right = str.length() - 1;
+
+        while (right > left) {
+            char a = str.charAt(left);
+            char b = str.charAt(right);
+
+            if (a >= 'A' && a <= 'Z') {
+                a = (char) (a + 32);
+            }
+
+            if (b >= 'A' && b <= 'Z') {
+                b = (char) (b + 32);
+            }
+
+            if (!((a >= 'a' && a <= 'z') || (a >= '0' && a <= '9'))) {
+                left++;
+                continue;
+            }
+
+            if (!((b >= 'a' && b <= 'z') || (b >= '0' && b <= '9'))) {
+                right--;
+                continue;
+            }
+
+            if (a != b) {
```

Comment:
Anh không yêu so sánh IgnoreCase nhưng nếu muốn thì chỉ cần thế này thôi


```suggestion
            if (toUpperCase(a) != toUpperCase(b)) {
```

thêm 

```java
import static java.lang.Character.toUpperCase;
```